### PR TITLE
docs(style guide): new section about internal links

### PR DESCRIPTION
**Pull request by muescha**
## Description

changes:
- new section about internal links
- fix internal links in example code
- add example for `localhost` links

_PS: i ask for help with wording, because i am not native english speaker_

### CODE REVIEW

File: docs/contributing/gatsby-style-guide.md [hunk 1]
```diff
@@ -126,19 +126,45 @@ Hyperlinks should contain the clearest words to indicate where the link will lea
 ```markdown
 <!-- Good -->
 
-[Gatsby's docs](https://www.gatsbyjs.org/docs/)
+[Gatsby's docs](/docs/)
```
Comment (laurieontech) [inline docs/contributing/gatsby-style-guide.md]: I'd remove this change to the example. While this uses .org and links should be relative, the example is about hyperlinks themselves and the text that accompanies them. By removing the full URL it's harder to tell that it is a hyperlink.

File: docs/contributing/gatsby-style-guide.md [hunk 2]
```diff
@@ -126,19 +126,45 @@ Hyperlinks should contain the clearest words to indicate where the link will lea
...
 
 <!-- Bad -->
 
-[here](https://www.gatsbyjs.org/docs/ "Gatsby's docs")
+[here](/docs/ "Gatsby's docs")
```
Comment (laurieontech) [inline docs/contributing/gatsby-style-guide.md]: Same note as above, let's keep these as is for now.

File: docs/contributing/gatsby-style-guide.md [hunk 3]
```diff
@@ -126,19 +126,45 @@ Hyperlinks should contain the clearest words to indicate where the link will lea
...
 ```
 
 In tutorials that are meant for beginners, use as few hyperlinks as possible to minimize distractions. In docs, it's ok to include as many hyperlinks as necessary to provide relevant and interesting information and resources.
 
+### Use relative hyperlinks for local links
+
+Hyperlinks which point to the same domain should have relative path (not include the domain). This is usefully when you test the build of the website on a local installations or on preview builds from your build system.
```
Comment (laurieontech) [inline docs/contributing/gatsby-style-guide.md]:
```suggestion
When referencing another page within [gatsbyjs.org](https://www.gatsbyjs.org/) hyperlinks should use relative paths (not include the full domain). This guarantees that all links function when running locally or in preview.
```

File: docs/contributing/gatsby-style-guide.md [hunk 4]
```diff
@@ -126,19 +126,45 @@ Hyperlinks should contain the clearest words to indicate where the link will lea
...
+<!-- Good -->
+
+[Gatsby's glossay](/docs/glossary)
+
+<!-- Bad -->
+
+[Gatsby's glossay](https://www.gatsbyjs.org/docs/glossary)
+```
+
+Note: Links to the Gatsby Cloud / Gatsby Inc. are at located at [gatsbyjs.com](https://www.gatsbyjs.com/) and should be absolute an path (include the domain). See also [Referencing Gatsby Cloud](#referencing-gatsby-cloud)
```
Comment (laurieontech) [inline docs/contributing/gatsby-style-guide.md]:
```suggestion
Note: Links to Gatsby Cloud/Gatsby Inc. are located at [gatsbyjs.com](https://www.gatsbyjs.com/) and should be referenced using an absolute path (domain included). See also [Referencing Gatsby Cloud](#referencing-gatsby-cloud)
```

File: docs/contributing/gatsby-style-guide.md (hunk 1 above)
Comment (muescha) [reply docs/contributing/gatsby-style-guide.md]: agree, longer url is more clear, 
i suggest link to https://www.gatsbyjs.com/cloud/

File: docs/contributing/gatsby-style-guide.md (hunk 3 above)
Comment (muescha) [reply docs/contributing/gatsby-style-guide.md]: done

File: docs/contributing/gatsby-style-guide.md (hunk 4 above)
Comment (muescha) [reply docs/contributing/gatsby-style-guide.md]: done

File: docs/contributing/gatsby-style-guide.md (hunk 1 above)
Comment (muescha) [reply docs/contributing/gatsby-style-guide.md]: changed to a full link, if you like more the https://www.gatsbyjs.org/docs/ then i can revert to it.

Comment (laurieontech) [reply docs/contributing/gatsby-style-guide.md]: I think that is a great choice!

**Pull request merged**